internal/kldrest/receiptstore.go

```diff
@@ -115,10 +115,10 @@ func (r *receiptStore) processReply(msgBytes []byte) {
 
 	// Insert the receipt into MongoDB - captures errors
 	if requestID != "" && r.persistence != nil {
-		if err := r.persistence.AddReceipt(&parsedMsg); err != nil {
-			log.Errorf("Failed to insert '%s' into receipt store: %+v", parsedMsg, err)
+		if err := r.persistence.AddReceipt(requestID, &parsedMsg); err != nil {
+			log.Errorf("%s: Failed to insert '%s' into receipt store: %+v", requestID, parsedMsg, err)
```

Comment:
Think we should probably panic here @vdamle - what do you think?

Reply:
I missed this, sorry. Will add that